src/foam/u2/view/RichTitledArrayView.js

```diff
@@ -0,0 +1,157 @@
+                console.log(row.value$);
+                this
+                  .startContext({ data: row })
+                    .start()
+                        .addClass(self.myClass('value-view-container')).style({
+                            borderColor: !isCollapsed && self.theme.primary5
+                        })
+                        .start(self.Cols)
+                            .addClass(self.myClass('header-row'))
+                            .start().style({alignContent: 'center'}).addClass(self.myClass('item-row'))
+                                .start('span').addClass(self.myClass('item-index')).add(i+1).end()
+                                .start('span').addClass(self.myClass('item-name')).add(
+                                row.value$.dot('label').map(label => label ? label : 'New Property')
+                                ).end()
+                            .end()
+                            .start().addClass(self.myClass('actions-holder'))
+                              .start().addClass(self.myClass('typeLabel'))
+                                .addClass(row.value$.dot('type').dot('name').map(name => name ? name.toLowerCase().replace(/[^a-z0-9_-]/g, '-') : ''))
+                                .add(row.value$.dot('type').dot('label').map(label => label))
+                              .end()
+                                .tag(self.Row.REMOVE, {
+                                    buttonStyle: 'TERTIARY',
+                                    themeIcon: 'trash',
+                                    label: '',
+                                })
+                                .start(foam.u2.tag.Button, {
+                                    buttonStyle: 'BLACK',
+                                    themeIcon: 'dropdown',
+                                }).addClass(self.myClass('control')).enableClass('opened', !isCollapsed).on('click', function() { self.toggleCollapse(i); })
+                                .end()
+                            .end()
+                        .end()
+                        .callIf(!isCollapsed, function() {
+                            this.start(valueView, { data$: row.value$ })
+                                .addClass(self.myClass('value-view'))
+                        .end();
+                        })
+                    .end()
+                .endContext();
+                row.onDetach(row.sub(self.updateDataWithoutFeedback));
+              });
+        }))
+        .start().addClass(self.myClass('add-row-container'))
+            .startContext({ data: this })
+                .tag(this.ADD_ROW, {
+                size: 'SMALL',
+                buttonStyle: 'SECONDARY',
+                label: 'Add New'
+                })
+            .endContext()
+        .end();
+    },
+  ],
+
+  listeners: [
+    function toggleCollapse(idx) {
+      this.collapsed[idx] = ! this.collapsed[idx];
+      this.collapsed = Object.assign({}, this.collapsed);
```

Comment:
```suggestion
      this.collapsed$set(idx, ! this.collapsed[idx])
```
This should trigger the slot